File: SteganographyApp.java
import javax.swing.*;
import java.awt.*;
import java.awt.event.*;
import java.awt.image.BufferedImage;
import javax.imageio.ImageIO;
import java.io.*;

public class SteganographyApp extends JFrame {

    private JLabel imageLabel;
    private JTextField messageField;
    private BufferedImage image;

    public SteganographyApp() {
        setTitle("Java Steganography App");
        setSize(700, 500);
        setDefaultCloseOperation(EXIT_ON_CLOSE);
        setLayout(new FlowLayout());

        JButton loadButton = new JButton("Load Image");
        JButton encodeButton = new JButton("Encode & Save");
        JButton decodeButton = new JButton("Decode Message");

        messageField = new JTextField(30);
        imageLabel = new JLabel();

        add(loadButton);
        add(new JLabel("Message:"));
        add(messageField);
        add(encodeButton);
        add(decodeButton);
        add(imageLabel);

        // Event Handlers
        loadButton.addActionListener(e -> loadImage());
        encodeButton.addActionListener(e -> encodeImage());
        decodeButton.addActionListener(e -> decodeImage());
    }

    private void loadImage() {
        JFileChooser chooser = new JFileChooser();
        if (chooser.showOpenDialog(this) == JFileChooser.APPROVE_OPTION) {
            try {
                File file = chooser.getSelectedFile();
                image = ImageIO.read(file);
                imageLabel.setIcon(new ImageIcon(image.getScaledInstance(200, 200, Image.SCALE_SMOOTH)));
            } catch (IOException e) {
                JOptionPane.showMessageDialog(this, "Error loading image.");
            }
        }
    }

    private void encodeImage() {
        if (image == null) {
            JOptionPane.showMessageDialog(this, "Please load an image first.");
            return;
        }

        String message = messageField.getText();
        if (message.isEmpty()) {
            JOptionPane.showMessageDialog(this, "Message cannot be empty.");
            return;
        }

        BufferedImage stegoImage = SteganographyProcessor.encode(image, message);
        try {
            File output = new File("encoded_image.png");
            ImageIO.write(stegoImage, "png", output);
            JOptionPane.showMessageDialog(this, "Message encoded and saved as 'encoded_image.png'.");
        } catch (IOException e) {
            JOptionPane.showMessageDialog(this, "Error saving image.");
        }
    }

    private void decodeImage() {
        if (image == null) {
            JOptionPane.showMessageDialog(this, "Please load an image first.");
            return;
        }

        String decodedMessage = SteganographyProcessor.decode(image);
        JOptionPane.showMessageDialog(this, "Decoded Message:\n" + decodedMessage);
    }

    public static void main(String[] args) {
        SwingUtilities.invokeLater(() -> new SteganographyApp().setVisible(true));
    }
}

File: steganography_app.py
import tkinter as tk
from tkinter import filedialog, messagebox
from PIL import Image
from steganography_processor import encode, decode

class SteganographyApp:
    def __init__(self, root):
        self.root = root
        self.root.title("Python Steganography App")
        self.root.geometry("600x400")

        self.image = None

        tk.Button(root, text="Load Image", command=self.load_image).pack(pady=5)

        tk.Label(root, text="Message:").pack()
        self.message_entry = tk.Entry(root, width=40)
        self.message_entry.pack(pady=5)

        tk.Button(root, text="Encode & Save", command=self.encode_image).pack(pady=5)
        tk.Button(root, text="Decode Message", command=self.decode_image).pack(pady=5)

    def load_image(self):
        file_path = filedialog.askopenfilename()
        if file_path:
            self.image = Image.open(file_path)
            messagebox.showinfo("Success", "Image Loaded Successfully!")

    def encode_image(self):
        if self.image is None:
            messagebox.showerror("Error", "Please load an image first.")
            return

        message = self.message_entry.get()
        if not message:
            messagebox.showerror("Error", "Message cannot be empty.")
            return

        encoded_image = encode(self.image, message)
        encoded_image.save("encoded_image.png")
        messagebox.showinfo("Success", "Message encoded and saved as encoded_image.png")

    def decode_image(self):
        if self.image is None:
            messagebox.showerror("Error", "Please load an image first.")
            return

        decoded_message = decode(self.image)
        messagebox.showinfo("Decoded Message", decoded_message)

if __name__ == "__main__":
    root = tk.Tk()
    app = SteganographyApp(root)
    root.mainloop()